### PR TITLE
Simpler handling surrogate pairs

### PR DESCRIPTION
**Pull request by KentoMoriwaki**
Simpler and more performant impl of https://github.com/dmonad/lib0/pull/45

Trade-off: This method does not guarantee behavior when the argument has an incorrect surrogate pair

### CODE REVIEW

**Comment by Young6118** [comment]
Good job!

File: diff.js [hunk 1]
```diff
@@ -41,12 +44,18 @@ import { equalityStrict } from './function.js'
 export const simpleDiffString = (a, b) => {
   let left = 0 // number of same characters counting from left
   let right = 0 // number of same characters counting from right
+  let inSurrogate = false
   while (left < a.length && left < b.length && a[left] === b[left]) {
+    inSurrogate = highSurrogateRegex.test(a[left])
```
Comment (dmonad) [inline diff.js:49]: Is it necessary to test this in every iteration? Couldn't we perform all the checks at the very end of the function and then adjust `left` & `right` if necessary?

**Review by dmonad — CHANGES_REQUESTED** [review]
I think I like this solution. Can you please adjust it a bit so we don't have to perform many unnecessary checks? 

Ideally, we can only perform the check once at the end of the function. There should be a small comment that describes the issue that the surrogate checks solve (i.e. refer to the ticket)

**Comment by KentoMoriwaki** [comment]
@dmonad I fixed and updated specs to improve coverages.

**Review by dmonad — APPROVED** [review]
Thank you, this looks great!

I'm going to publish a new release today.

**Comment by tommoor** [comment]
The link in the original PR suggests that this would fix https://github.com/yjs/yjs/issues/303 however the behavior doesn't seem to have changed in regards to those original reproduction instructions.

**Comment by KentoMoriwaki** [comment]
@tommoor  Are you certain you're using `lib0@0.2.69` in your environment? I have verified that this issue has been resolved in mine.

**Pull request merged**